eq-author-api/schema/tests/commentsConfirmationPage.test.js

```diff
@@ -0,0 +1,242 @@
+  it("should edit a comment on confirmationPage and then query that comment", async () => {
+    const newComment = await createComment(ctx, {
+      confirmationId: confirmationId,
+      commentText: "a new comment is created",
+    });
+    const commentId = newComment.id;
+
+    const queryEditedComment = await updateComment(ctx, {
+      confirmationId: confirmationId,
+      commentId: commentId,
+      commentText: "an edited comment",
+    });
+
+    expect(queryEditedComment).toMatchObject({
+      id: commentId,
+      commentText: "an edited comment",
+      editedTime: expect.any(String),
+    });
+  });
+
+  it("should delete a comment on confirmationPage", async () => {
+    const newComment = await createComment(ctx, {
+      confirmationId: confirmationId,
+      commentText: "a new comment is created",
+    });
+
+    const queriedComment = await deleteComment(ctx, {
+      confirmationId: confirmationId,
+      commentId: newComment.id,
+    });
+
+    expect(queriedComment.confirmationPage.comments).toHaveLength(0);
+    expect(queriedComment).toMatchObject({
+      confirmationPage: {
+        id: confirmationId,
+        comments: [],
+      },
+    });
+  });
+
+  it("should create a comment object if Questionnaire doesn't have one", async () => {
+    //creating a questionnaire outside resolvers to ensure no existing comment object...
+    const questionnaire = await createQuestionnaire(ctx.questionnaire, ctx);
+
+    const comment = await queryComments(ctx, {
+      confirmationId: questionnaire.sections[0].pages[0].confirmation.id,
+    });
+
+    expect(comment.confirmationPage).toMatchObject({
+      id: expect.any(String),
+      comments: expect.any(Array),
+    });
+  });
+
+  describe("replies", () => {
+    it("should add a reply to a comment - on confirmationPage", async () => {
+      const comment = await createComment(ctx, {
+        confirmationId: confirmationId,
+        commentText: "a new comment is created",
+      });
+
+      await createReply(ctx, {
+        confirmationId: confirmationId,
+        commentId: comment.id,
+        commentText: "a new reply is created",
+      });
+
+      const queriedComment = await queryComments(ctx, {
+        confirmationId: confirmationId,
+      });
+
+      expect(
+        queriedComment.confirmationPage.comments[0].replies
+      ).toMatchObject([{ commentText: "a new reply is created" }]);
+    });
+
+    it("should edit a reply on confirmationPage", async () => {
+      const newComment = await createComment(ctx, {
+        confirmationId: confirmationId,
+        commentText: "a new comment is created",
+      });
+      const commentId = newComment.id;
+
+      const reply = await createReply(ctx, {
+        confirmationId: confirmationId,
+        commentId: commentId,
+        commentText: "a new reply is created",
+      });
+
+      const replyId = reply.id;
+
+      const editedReply = await updateReply(ctx, {
+        confirmationId: confirmationId,
```

Comment:
You could reduce this object by doing the following

```suggestion
        confirmationId,
```